rest/model/dns/tsig_key.go

```diff
@@ -0,0 +1,18 @@
+package dns
+
+// TSIGKey wraps an NS1 /tsig resource
+type TSIGKey struct {
+	Name      string `json:"name,omitempty"`
+	Algorithm string `json:"algorithm,omitempty"`
+	Secret    string `json:"secret,omitempty"`
+}
+
+// NewTSIGKey takes a name, algorithm and secret and creates a new TSIG key.
+func NewTsigKey(name string, algorithm string, secret string) *TSIGKey {
+	tsig_key := TSIGKey{
```

Comment:
No underscores in var names. Please run a go linter on this code, it should point out stuff like this.